### ISSUE TITLE
Support for "delete" of targets missing in the source

### ISSUE THREAD

**Issue by andrus** [nhl/link-etl#23]
as of 1.1 we only support "create-or-update" synchronization. Need to start supporting "delete" as well.

**Comment by atomashpolskiy** [comment]
Have thought on this a bit. Probably we'll need several strategies for deletion:
1) Collect all business keys for created and updated entities; then, as the last phase of ETL task:
a) either delete all entities from target that have non-matching keys,
b) or mark them as inactive (via some column); this would also require marking all matching entities as active throughout the ETL process
In the latter case, some extra configuration will be needed: name of the column, type and values for active/inactive
2) As the first phase of ETL task, truncate target
3) Some "no-action/callback" strategy to let the user define more complex processing of deletions (e.g. when date-time intervals are used for marking entries as active - active_since etc.)

Questions:
1) Possible ways of enabling deletion? Via extractor configuration or ETL task builder parameters?
2) Callbacks on per-entity deletion (same as for create/update)? Not applicable for option 2.

**Comment by andrus** [comment]
After #26 was implemented and our pipeline was explicitly exposed in the code, it became much easier to design new pipelines. So some analysis:

1. Generally we can not reuse the extractor used by CreateOrUpdateTask for delete (and can't attach delete operation to CreateOrUpdateTask). Especially if the task uses SyncToken. As in this case we are not getting all sources and can not reliably decide whether a given record needs to be deleted.
2. DeleteTask will be a separate task and will use its own extractor (in simple cases like "SELECT * FROM my_table" it may reuse CreateOrUpdateTask extractor).
3. DeleteTask pipeline will somewhat the opposite to CreateOrUpdateTask:

    BatchRunner.iterate_over_all_targets // allow target filtering?
        for_each_segment
             mapTargets
             matchSource
             deleteMissingTargets
             commit
